cli/src/upload.rs

```diff
@@ -389,7 +395,10 @@ fn handle_xcresult(
                 .map_err(|e| anyhow::anyhow!("Failed to write junit file: {}", e))?;
             let junit_temp_path_str = junit_temp_path.to_str();
             if let Some(junit_temp_path_string) = junit_temp_path_str {
-                temp_paths.push(junit_temp_path_string.to_string());
+                temp_paths.push(JunitReportFileWithStatus {
+                    junit_path: junit_temp_path_string.to_string(),
+                    status: None,
+                });
```

Comment:
nit: now that we've implemented `From<String>` for `JunitReportFileWithStatus`, this simplifies to the following:
```suggestion
                temp_paths.push(junit_temp_path_string.into());
```